code/SoloSavingsApp/src/main/java/com/soloSavings/controller/TransactionController.java

```diff
@@ -76,5 +96,35 @@ public ResponseEntity<?> getThisMonthIncome(@PathVariable("user_id") Integer use
         } catch (TransactionException e) {
             return new ResponseEntity<>(e.getMessage(),HttpStatus.NO_CONTENT);
         }
+        
+    }
+    @GetMapping("/{user_id}/export/csv")
+    public ResponseEntity<Resource> exportTransactionsToCsv(@PathVariable("user_id") Integer userId) throws IOException {
```

Comment:
I was thinking maybe we could have the business logic in the service layer and keep controller layer purely for REST call?